### DISCUSSION TITLE
EAVアンチパターンとは？

### DISCUSSION THREAD

**Discussion by yuyakawai0411** [kakeru-one/web-engineering-roundtable#44] [Ideas]
## 前提条件
RDB(リレーショナルデータベース)は「同種のデータの大量処理」のためのシステムである。ゆえに、カラムの意味が同じパターンで厳密に定義されていなければならない。

## EAVアンチパターンとは
可変属性をサポートするために柔軟な設計をしようとするアンチパターン。
例えば、エラーデータはエラーの種類によって取り扱う属性が可変する。これをRDBで取り扱うため、attr_nameとattr_valueのように値の属性名と値そのものを格納する2つのカラムを作成し、可変属性をサポートできるようにする。こうしてしまうと、カラムが複数の意味を持つことになり、RDBの設計思考と外れてしまう。

## 対策
MySQLで可変属性を扱うときは、JSON形式で構造化し、JSONデータ型で保存すればいいと思った。-[参考文献](https://zenn.dev/rescuenow/articles/854cf493737672)
エラーログの出力もJSON形式で構造化されて出力されるため。

**Comment by kakeru-one** [comment]
これ、SQLアンチパターンとして解説されてたりします。
https://qiita.com/skyc_lin/items/37365a36416d0dc42431
